### PR TITLE
chore(instrumentation): move instrumentation configuration from opentelemetry-operator

### PR DESCRIPTION
**Pull request by mat-rumian**
:warning: `BREAKING-CHANGE` :warning:

PR moves out `instrumentation` related configuration from `opentelemetry-operator` values.

#### Moved:
* From `opentelemetry-operator.instrumentationJobImage` to `instrumentation.instrumentationJobImage`
* From `opentelemetry-operator.createDefaultInstrumentation` to `instrumentation.createDefaultInstrumentation`
* From `opentelemetry-operator.instrumentationNamespaces` to `instrumentation.instrumentationNamespaces`
* From `opentelemetry-operator.instrumentation.dotnet.traces` to `instrumentation.dotnet.traces`
* From `opentelemetry-operator.instrumentation.dotnet.metrics` to `instrumentation.dotnet.metrics`
* From `opentelemetry-operator.instrumentation.dotnet.extraEnvVars` to `instrumentation.dotnet.extraEnvVars`
* From `opentelemetry-operator.instrumentation.java.traces` to `instrumentation.java.traces`
* From `opentelemetry-operator.instrumentation.java.metrics` to `instrumentation.java.metrics`
* From `opentelemetry-operator.instrumentation.java.extraEnvVars` to `instrumentation.java.extraEnvVars`
* From `opentelemetry-operator.instrumentation.nodejs` to `instrumentation.nodejs`
* From `opentelemetry-operator.instrumentation.python.traces` to `instrumentation.python.traces`
* From `opentelemetry-operator.instrumentation.python.metrics` to `instrumentation.python.metrics`
* From `opentelemetry-operator.instrumentation.python.extraEnvVars` to `instrumentation.python.extraEnvVars`

#### Changed:
* From `opentelemetry-operator.instrumentation.dotnet.repository` to `opentelemetry-operator.autoInstrumentationImage.dotnet.repository`
* From `opentelemetry-operator.instrumentation.dotnet.tag` to `opentelemetry-operator.autoInstrumentationImage.dotnet.tag`
* From `opentelemetry-operator.instrumentation.java.repository` to `opentelemetry-operator.autoInstrumentationImage.java.repository`
* From `opentelemetry-operator.instrumentation.java.tag` to `opentelemetry-operator.autoInstrumentationImage.java.tag`
* From `opentelemetry-operator.instrumentation.nodejs.repository` to `opentelemetry-operator.autoInstrumentationImage.nodejs.repository`
* From `opentelemetry-operator.instrumentation.nodejs.tag` to `opentelemetry-operator.autoInstrumentationImage.nodejs.tag`
* From `opentelemetry-operator.instrumentation.python.repository` to `opentelemetry-operator.autoInstrumentationImage.python.repository`
* From `opentelemetry-operator.instrumentation.python.tag` to `opentelemetry-operator.autoInstrumentationImage.python.tag`

#### Deleted:
* `opentelemetry-operator.instrumentation.dotnet.image`
* `opentelemetry-operator.instrumentation.java.image`
* `opentelemetry-operator.instrumentation.nodejs.image`
* `opentelemetry-operator.instrumentation.python.image`

### Checklist

- [x] Changelog updated or skip changelog label added
- [ ] Documentation updated
- [ ] Template tests added for new features
- [ ] Integration tests added or modified for major features

### CODE REVIEW

**Comment by mat-rumian** [comment]
Do not merge until PRs with documentation will be ready.

File: .changelog/3733.breaking.txt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+* From `opentelemetry-operator.instrumentation.dotnet.extraEnvVars` to `instrumentation.dotnet.extraEnvVars`
+* From `opentelemetry-operator.instrumentation.java.traces` to `instrumentation.java.traces`
+* From `opentelemetry-operator.instrumentation.java.metrics` to `instrumentation.java.metrics`
+* From `opentelemetry-operator.instrumentation.java.extraEnvVars` to `instrumentation.java.extraEnvVars`
+* From `opentelemetry-operator.instrumentation.nodejs` to `instrumentation.nodejs`
+* From `opentelemetry-operator.instrumentation.python.traces` to `instrumentation.python.traces`
+* From `opentelemetry-operator.instrumentation.python.metrics` to `instrumentation.python.metrics`
+* From `opentelemetry-operator.instrumentation.python.extraEnvVars` to `instrumentation.python.extraEnvVars`
+
+#### Changed:
```
Comment (sumo-drosiek) [inline .changelog/3733.breaking.txt:18]: What is the difference between `moved` and `changed`?

**Review by sumo-drosiek — APPROVED** [review]

File: .changelog/3733.breaking.txt (hunk 1 above)
Comment (mat-rumian) [reply .changelog/3733.breaking.txt:18]: Moved are the keys `moved` from `opentelemetry-operator` key to `instrumentation`. Changed are keys which stay in the `opentelemetry-operator` but in different place (if it makes sense :))

Comment (sumo-drosiek) [reply .changelog/3733.breaking.txt:18]: It makes sense, but not sure if the distinguishment is significant and needed for customer

**Review by swiatekm-sumo — COMMENTED** [review]
The change looks good to me, I'm wondering how we should document it though. In practice, a lot of users unfortunately copy the entirety of our values.yaml, so this will block their upgrade even if they don't have tracing enabled.

I think we should:
1. Add a section to the Troubleshooting documentation in help.sumologic.com specifically for this change.
2. Try to figure out if we can show a custom error message pointing to that documentation.

**Comment by mat-rumian** [comment]
@swiatekm-sumo 
> 1. Add a section to the Troubleshooting documentation in help.sumologic.com specifically for this change.
I've added a section in Troubleshooting documentation please see - https://github.com/SumoLogic/sumologic-documentation/pull/4197

> 2. Try to figure out if we can show a custom error message pointing to that documentation.
I was checking if it's possible but validation happens as first (even if I add some checks for specific keys in affected templates).

**Review by swiatekm-sumo — APPROVED** [review]

**Pull request merged**